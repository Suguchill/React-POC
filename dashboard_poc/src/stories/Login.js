import React from 'react';
import PropTypes from 'prop-types';
import './login.css';
import Input from '../components/atoms/Input/Input';
import Button from '../components/atoms/Button/Button';
import 'bootstrap/dist/css/bootstrap.min.css';
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faEye } from "@fortawesome/free-solid-svg-icons";
const eye = <FontAwesomeIcon icon={faEye} />;

export const Login = ({value1, value2, emailErr, passErr, passwordShown, formValid, loginErrMsg, boxShadowColor}) => {
    return(
    <div className="bgColorSet">
    <div className="container"><br />
        <div className="row justify-content-center">
            <div className="col-md-5 col-sm-12 col-12 login-wrapper">
                <div className="card login_card">
                    <div className="card-body">
                        <h4 className="text-left">ACCOUNT LOGIN</h4>
                        <form className="mt-4">
                            <div className="form-group">
                                <label>USERNAME</label>
                                <Input type="text" className="form-control login-input" style={{boxShadow: boxShadowColor?"0px 1px 9px -3px "+boxShadowColor:'none'}} value={value1}/>
                            </div>
                            {emailErr &&
                                <div className="mb-2">
                                    <span className="text-danger">{emailErr}</span>
                                </div>
                            }
                            <div className="form-group">
                                <label>PASSWORD</label>
                                <div className="password-wrapper">
                                    <Input type={passwordShown ? "text" : "password"} className="form-control login-input" style={{boxShadow: boxShadowColor?"0px 1px 9px -3px "+boxShadowColor:'none'}} value={value2}/>
                                    <i>{eye}</i>
                                </div>
                            </div>
                            {passErr &&
                                <div className="mb-2">
                                    <span className="text-danger">{passErr}</span>
                                </div>
                            }
                            <div className="login-bottom-wrapper">
                                <div className="rememberMe-wrapper"><Input type="checkbox" id="rememberCheck" />
                                    <label className="ml-2">Remember me</label>
                                </div>
                                <div><Button className="" value="Forgot Password?" color="black" href="#" /></div>
                            </div>
                            <div className="mt-4">
                                <Button className="btn btn-dark" value="Login" disabled={!formValid} />
                            </div>
                            <div className="mt-3">
                                <span className="text-danger"><p>{loginErrMsg}</p></span>
                            </div>
                        </form>
                    </div>
                </div>
            </div>
        </div>
        </div>
        </div>
)
};

Login.propTypes = {
    value1:PropTypes.string,
    value2:PropTypes.string,
    emailErr: PropTypes.string,
    passErr: PropTypes.string,
    passwordShown: PropTypes.bool,
    formValid: PropTypes.bool,
    loginErrMsg: PropTypes.string,
    boxShadowColor: PropTypes.string
};

Login.defaultProps = {
    value1:"username",
    value2:"password",
    emailErr: "",
    passErr: "",
    passwordShown: false,
    formValid: false,
    loginErrMsg: "",
    boxShadowColor: ""
};