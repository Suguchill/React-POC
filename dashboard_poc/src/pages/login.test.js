import React from "react";
import Login from './login';
import { render } from "@testing-library/react"
import userEvent from '@testing-library/user-event'
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';
configure({adapter: new Adapter()});

describe('loginComponent', () => {
    const setLoginState = jest.fn();
    const LoginState = jest.spyOn(React, "useState");
    LoginState.mockImplementation(loginState => [loginState, setLoginState]);


    it('should render correctly ', () => {
        const component = shallow(<Login debug />);
        expect(component).toMatchSnapshot();
    });

    it("testing event handlers",() => {
        const { getByText, getByTestId} = render(<Login setLoginState={setLoginState} />)
        
        const name = getByTestId("userField")
        userEvent.type(name, 'admin')
        const pass = getByTestId("passField")
        userEvent.type(pass, '')

        getByText('*Required')
    })

});