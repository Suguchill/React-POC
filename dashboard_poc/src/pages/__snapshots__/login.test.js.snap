// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loginComponent should render correctly  1`] = `ShallowWrapper {}`;
